ios/Classes/SwiftFlutterVlcPlayerPlugin.swift

```diff
@@ -0,0 +1,284 @@
+    private var channel: FlutterMethodChannel
+    private var eventChannel: FlutterEventChannel
+    private var player: VLCMediaPlayer
+    private var eventChannelHandler: VLCPlayerEventStreamHandler
+    private var aspectSet = false
+    
+    
+    
+    public init(withFrame frame: CGRect, withRegistrar registrar: FlutterPluginRegistrar, withId id: Int64){
+        self.registrar = registrar
+        self.hostedView = UIView()
+        self.player = VLCMediaPlayer()
+        self.channel = FlutterMethodChannel(name: "flutter_video_plugin/getVideoView_\(id)", binaryMessenger: registrar.messenger())
+        self.eventChannel = FlutterEventChannel(name: "flutter_video_plugin/getVideoEvents_\(id)", binaryMessenger: registrar.messenger())
+        self.eventChannelHandler = VLCPlayerEventStreamHandler()
+        
+        
+    }
+    
+    public func view() -> UIView {
+        channel.setMethodCallHandler({
+            [weak self] (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
+            
+            guard let self = self else { return }
+            
+            if let arguments = call.arguments as? Dictionary<String,Any>
+            {
+                switch(FlutterMethodCallOption(rawValue: call.method)){
+                case .initialize:
+                    
+                    guard let  urlString = arguments["url"] as? String, let url = URL(string: urlString) else {
+                        result(FlutterError(code: "500",
+                                            message: "Url is need to initialization",
+                                            details: nil)
+                        )
+                        return
+                    }
+                    
+                    let media = VLCMedia(url: url)
+                    
+                    
+                    self.player.media = media
+                    self.player.position = 0.5
+                    self.player.drawable = self.hostedView
+                    self.player.delegate = self.eventChannelHandler
+                    
+                    result(nil)
+                    return
+                case .setPlaybackState:
+                    let playbackState = arguments["playbackState"] as? String
+                    
+                    if (playbackState == "play") {
+                        self.player.play()
+                    } else if (playbackState == "pause") {
+                        self.player.pause()
+                    } else if (playbackState == "stop") {
+                        self.player.stop()
+                    }
+                    
+                    result(nil)
+                    return
+                case .dispose:
+                    self.player.stop()
+                    return
+                case .changeURL:
+                    self.player.stop()
+                    let url = arguments["url"] as? String
+                    let media = VLCMedia(url: URL(string: url!)!)
```

Comment:
@mitchross what do you think about making the same guard statement you did on the `.initialize` case?